types/stylelint/index.d.ts

```diff
@@ -110,7 +110,7 @@ declare namespace stylelint {
 		_processorFunctions?: Map<string, ReturnType<Processor>['postprocess']>;
 		allowEmptyInput?: boolean;
 		cache?: boolean;
-		fix?: boolean;
+		fix?: boolean | 'lax' | 'strict';
```

Comment:
[suggestion] The following internal properties may make the implementation a bit more understandable by reducing repetitions:

```suggestion
		fix?: boolean | 'lax' | 'strict';
		/** @internal */
		_fixLax?: boolean;
		/** @internal */
		_fixStrict?: boolean;
```

E.g.

```diff
-	const fix = stylelint._options.fix === 'lax' || stylelint._options.fix === true;
 	const syntax = await (customSyntax
 		? getCustomSyntax(customSyntax, stylelint._options.configBasedir)
-		: cssSyntax(fix));
+		: cssSyntax(stylelint._options._fixLax));
```

Reply:
Instead of a normalized autofix mode in a hidden field, it may be better to add a utility function to normalize it. For example, the function might be as follows:

```js
export function normalizeFixMode(options, config) {
  // Check options first

  // If options have no fix mode, check config
}
```

I want to combine checks like `fix === 'lax'` in one place as much as possible.

Reply:
Thanks for all the suggestions! I agree that adding a utility function would make the implementation cleaner.
I'll take care of this later.

Reply:
Applied in [this commit](https://github.com/stylelint/stylelint/pull/8106/commits/f59eade747442f97d3257ad85954c20be8badbcb).